collector/meminfo_linux.go

```diff
@@ -89,6 +121,16 @@ func parseMemInfo(r io.Reader) (map[string]float64, error) {
 	for scanner.Scan() {
 		line := scanner.Text()
 		parts := strings.Fields(string(line))
+		key := ""
+
+		// /proc/meminfo format:
+		//   <metric name>: <metric val> [kB]
+		// /sys/devices/system/node/node%d/meminfo format:
+		//   Node <node number> <metric name>: <metric val> [kB]
+		if len(parts) >= 4 {
+			key = parts[0] + parts[1] + "_"
```

Comment:
It helps as it means that scrapes are stateless, if you're sharing metrics then you have races with concurrent scrapes.